### ISSUE TITLE
hiddenservices_client not working

### ISSUE THREAD

**Issue by qstokkink** [Tribler/gumby#286]
The [HiddenServicesClient](https://github.com/Tribler/gumby/blob/devel/experiments/dispersy/hiddenservices_client.py) seeder and downloader are using different infohashes, this causes the file not to download.

This _can_ be fixed by using a `TorrentDefNoMetainfo` for the downloader, using the same infohash as used by the seeder. I have [an example implementation for the downloader's TorrentDef](https://github.com/qstokkink/gumby/blob/5bd80f27e92e7bcd59685bd6ace8a7dd0687b643/experiments/distributed_tunnel_pooled/bbq_client.py#L106) on my branch for a __bbq__ ➡️  __DAS5__ experiment (so you __can't copy-paste__ my experiment).

@devos50 @pimveldhuisen 

_EDIT:_ Adding peers to the download is also required to get it back up and running, see https://github.com/qstokkink/gumby/tree/fix_hiddenseeding for current fix.

**Comment by synctext** [comment]
Wonderful news if this is the killer bug that has been haunting us for months! good work

**Comment by synctext** [comment]
@devos50 you briefly indicated that this was not the only root-cause-of-failure. So another issue is preventing the tunnel+multichain byte-counting experiment from properly functioning?

**Comment by devos50** [comment]
During our investigation, we found various issues where this was one of them. I'm pretty sure that we've printed the info hashes of a download to see whether they are equal and iirc, they were. @qstokkink are you also using hidden seeding?

The other issue is that a download bypasses the tunnels, making the download just plain. We suspect that this is something in libtorrent but I think this is a harder issue to solve and we might have to discuss this with the libtorrent developers.

So I think this absolutely contributes to the solution but there are still issues left that should be solved and investigated better.

**Comment by qstokkink** [comment]
Woops I just realized the experiment started working due to my peer sharing mechanism, which came to life after fixing the infohashes (so I attributed the fix to that instead). I also discovered it only does anonymous downloads and not anonymous seeding: double woops.

All my errors aside, my _anonymous downloading_ experiment __is working__ (no bypasses). For anyone desperately requiring an anonymous downloading experiment (@pimveldhuisen ?) the backport for the hiddenservices_client is here: https://github.com/qstokkink/gumby/tree/fix_hiddenseeding

Why my overkill solution works and the current experiment does not, is still an open issue. It has something to do with the peers being added in a certain way to the download.
